File: 02/solution.js
const fs = require('fs');

// score for each shape
const scores = {
  X: 1,
  Y: 2,
  Z: 3
};

// // win-loss outcomes for each pair of shapes
// const outcomes = {
//   AX: 0,
//   AY: 6,
//   AZ: 3,
//   BX: 6,
//   BY: 0,
//   BZ: 3,
//   CX: 3,
//   CY: 6,
//   CZ: 0
// };

// FIX: chatgpt can't seem to get these values right so i had to manually write these in
const outcomes = {AX: 0, AY: 6, AZ: 3, BX: 6, BY: 0, BZ: 3, CX: 3, CY: 6, CZ: 0 };

// read the strategy guide from the file
const strategy = fs.readFileSync('strategy_guide.txt', 'utf8').trim();

// split the strategy into rounds
const rounds = strategy.split('\n');

// calculate the total score
let totalScore = 0;
for (let round of rounds) {
  const [opponent, myChoice] = round.split(' ');
  const outcome = outcomes[opponent + myChoice];
  const myScore = scores[myChoice];
  totalScore += myScore + outcome;
}

console.log(totalScore);

File: 02/solution.py
import sys

file = open('input.txt', 'r')
data = file.read()
lines = data.split('\n')
lines.pop()

# dictionary to map letters to their corresponding numbers
mapping = {
    'A': 1,
    'B': 2,
    'C': 3,
    'X': 1,
    'Y': 2,
    'Z': 3
}

total_score = 0

for line in lines:
    # get the number of the shape chosen by your opponent and you
    opponent_shape = mapping[line[0]]
    my_shape = mapping[line[2]]

    # calculate the score of the round
    score = 0
    if my_shape == opponent_shape:
        score = 3 + my_shape
    elif (my_shape == 1 and opponent_shape == 3) or (my_shape == 2 and opponent_shape == 1) or (my_shape == 3 and opponent_shape == 2):
        score = 6 + my_shape
    else:
        score = 0 + my_shape

    # add the round score to the total score
    total_score += score
    
print(total_score)

file.close()
